### ISSUE TITLE
Bug in RuntimeWarning for _imaging

### ISSUE THREAD

**Issue by lovetox** [python-pillow/Pillow#2688]
### What did you do?
Install Pillow on Windows via pip in a MSYS2 enviroment
import image

### What actually happened?
```
RuntimeWarning: The _imaging extension was built for another version of Pillow or PIL: Core
Version: "4.2.1"Pillow Version:  4.2.1
```

### What versions of Pillow and Python are you using?
python 3.6

I think the check in https://github.com/python-pillow/Pillow/blob/master/PIL/Image.py#L57 fails because `"4.2.1"` != `4.2.1`

**Comment by hugovk** [comment]
You could put a print(type(...)) for both to check what they are. 

4.2.1 can't be a float, with two decimals :)

**Comment by lovetox** [comment]
they are both strings, one has quotes in it the other dont.

**Comment by wiredfool** [comment]
What's `sys.platform` in your python?  Is it `win32`?  Is there anything in `sys` that tags it to MSYS2?

**Comment by lovetox** [comment]
https://docs.python.org/3/library/sys.html#sys.platform

there is no value 'msys2'.


```
test@test MINGW32 ~
$ python3
Python 3.6.2rc1 (default, Jun 26 2017, 07:51:29)  [GCC 6.3.0 32 bit] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.platform
'win32'
>>>
```

i install pillow with pip3

```
$ pip3 install pillow
Collecting pillow
  Using cached Pillow-4.2.1.tar.gz
Requirement already satisfied: olefile in c:/msys64/mingw32/lib/python3.6/site-packages (from pillow)
Building wheels for collected packages: pillow
  Running setup.py bdist_wheel for pillow: started
  Running setup.py bdist_wheel for pillow: finished with status 'done'
  Stored in directory: C:/Users/Philipp/AppData/Local/pip/Cache/wheels/88/15/71/1970b5f85d14e373f3a71279ba7f6d3ced6a48179a5e8d44b3
Successfully built pillow
Installing collected packages: pillow
Successfully installed pillow-4.2.1
```

**Comment by lovetox** [comment]
The only way i found it this

sys.verison reports:


`3.6.2rc1 (default, Jun 26 2017, 07:51:29)  [GCC 6.3.0 32 bit]` in msys


`3.4.4 (v3.4.4:737efcadf5a6, Dec 20 2015, 19:28:18) [MSC v.1600 32 bit (Intel)]` installed from python.org

**Comment by wiredfool** [comment]
Msys2 is a strange beast.  I've spun up a ec2 instance, and I'm getting `sys.platform` == `msys`, but just installing python (3.6.1) and python3-pip doesn't give a working pip command, python3-setuptools doesn't give a working setuptools. Also don't see virtualenv anywhere. 

Python2 gets a working pip instance, but I can't find the jpeg libraries anywhere. So disabling them just to test this with: `python setup.py build_ext --disable-jpeg install`, I'm able to install and the version check works properly. 

So, there's something else going on with your setup.

**Comment by lovetox** [comment]
thats probably because you use the MSYS shell, but this is only used to update core packages i think

if you want to build software for windows, you usually use either

MSYS2 MinGW 32-bit
MSYS2 MinGW 64-bit

you should find shortcuts for that in your installation.

**Comment by wiredfool** [comment]
What msys2 packages do you have installed for the dependencies?

**Comment by lovetox** [comment]
Sorry, i realized in this conversations that its a bit more complicated as to just say "i use a msys env"

so to be more exact

i update core packages first like it is described on http://www.msys2.org/

then i start the MSYS2 MinGW 32-bit shell

`pacman -S mingw32/mingw-w64-i686-python3`
`pacman -S mingw32/mingw-w64-i686-python3-pip`
`pacman -S mingw32/mingw-w64-i686-python3-setuptools`
`pacman -S mingw-w64-i686-toolchain`

then just `pip3 install pillow`

i think thats it

**Comment by wiredfool** [comment]
Does that get you libz and libjpeg, with headers?

**Comment by lovetox** [comment]
hm im not sure

```
$ find . -name '*libjpeg*'
./home/Philipp/gajim-gtk3-fork/win/_build_root/mingw32/bin/libjpeg-8.dll
./home/Philipp/gajim-gtk3-fork/win/_build_root/var/lib/pacman/local/mingw-w64-i686-libjpeg-turbo-1.5.2-1
./home/Philipp/gtk-windows-bin/bin/libjpeg-7.dll
./mingw32/bin/libjpeg-8.dll
./mingw32/lib/libjpeg.a
./mingw32/lib/libjpeg.dll.a
./mingw32/lib/pkgconfig/libjpeg.pc
./mingw32/share/doc/libjpeg-turbo
./mingw32/share/doc/libjpeg-turbo/libjpeg.txt
./mingw32/share/licenses/libjpeg-turbo
./mingw64/bin/libjpeg-8.dll
./mingw64/lib/libjpeg.a
./mingw64/lib/libjpeg.dll.a
./mingw64/lib/pkgconfig/libjpeg.pc
./mingw64/share/doc/libjpeg-turbo
./mingw64/share/doc/libjpeg-turbo/libjpeg.txt
./mingw64/share/licenses/libjpeg-turbo
./var/cache/pacman/pkg/mingw-w64-i686-libjpeg-turbo-1.5.1-1-any.pkg.tar.xz
./var/cache/pacman/pkg/mingw-w64-i686-libjpeg-turbo-1.5.2-1-any.pkg.tar.xz
./var/cache/pacman/pkg/mingw-w64-x86_64-libjpeg-turbo-1.5.1-1-any.pkg.tar.xz
./var/lib/pacman/local/mingw-w64-i686-libjpeg-turbo-1.5.1-1
./var/lib/pacman/local/mingw-w64-x86_64-libjpeg-turbo-1.5.1-1
```

```
$ find . -name '*libz*'
./mingw32/lib/libz.a
./mingw32/lib/libz.dll.a
./mingw64/lib/libz.a
./mingw64/lib/libz.dll.a
```

**Comment by lovetox** [comment]
So could i modify that lines that tell what version goes into the compilation process? i dont care to be honest i just want to get it working.

if i try to install with
`python3 setup.py install`

```
from PIL import Image
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\msys64\mingw32\lib\python3.6\site-packages\pillow-4.3.0.dev0-py3.6-mingw.egg\PIL\Image.py", line 56, in <module>
ImportError: cannot import name '_imaging'
```

i read that is a common problem on windows, but i found no solution.

**Comment by wiredfool** [comment]
I think there's a fix in #2746.

**Comment by lovetox** [comment]
thanks that fixes the version problem
